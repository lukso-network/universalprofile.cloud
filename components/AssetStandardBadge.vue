<script setup lang="ts">
type Props = {
  asset?: Asset
}

const props = defineProps<Props>()
const isLoaded = computed(() => props.asset && !props.asset?.isLoading)
const standard = computed(() => props.asset?.standard)
const slots = useSlots()
const { formatMessage } = useIntl()
</script>

<template>
  <lukso-tag v-if="slots.default" size="x-small" background-color="lukso-90"
    ><slot
  /></lukso-tag>
  <div v-else-if="isLoaded" class="flex whitespace-nowrap">
    <lukso-tag
      v-if="isCollection(asset)"
      size="x-small"
      background-color="lukso-90"
      >{{
        formatMessage('asset_standard_badge_collection_contract')
      }}</lukso-tag
    >
    <lukso-tag
      v-else-if="standard"
      size="x-small"
      :background-color="
        standard === STANDARDS.UNKNOWN ? 'neutral-90' : 'lukso-90'
      "
      >{{ STANDARDS_ABBREVIATIONS[standard as Standard] }}</lukso-tag
    >
  </div>
  <AppPlaceholderLine v-else class="h-[20px] w-10" />
</template>
