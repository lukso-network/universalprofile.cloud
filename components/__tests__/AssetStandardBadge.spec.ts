import { renderSuspended } from '@nuxt/test-utils/runtime'
import { fireEvent, screen } from '@testing-library/vue'
import { beforeEach, describe, expect, it, vi } from 'vitest'

import AssetStandardBadge from '@/components/AssetStandardBadge.vue'

describe('AssetStandardBadge', () => {
  beforeEach(() => {
    useIntl().setupIntl(defaultConfig)
  })

  it('should render loading state', async () => {
    const component = await renderSuspended(AssetStandardBadge)

    expect(component.html()).toMatchSnapshot()
  })

  it('should render slot content', async () => {
    const component = await renderSuspended(AssetStandardBadge, {
      slots: {
        default: 'Slot content',
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render collection badge', async () => {
    const asset: Asset = {
      standard: 'LSP8IdentifiableDigitalAsset',
      tokenIdsData: [
        {
          tokenId: '0x1',
        },
      ],
    }
    const component = await renderSuspended(AssetStandardBadge, {
      props: {
        asset,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render LSP8 badge', async () => {
    const asset: Asset = {
      standard: 'LSP8IdentifiableDigitalAsset',
    }
    const component = await renderSuspended(AssetStandardBadge, {
      props: {
        asset,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })

  it('should render LSP7 badge', async () => {
    const asset: Asset = {
      standard: 'LSP7DigitalAsset',
    }
    const component = await renderSuspended(AssetStandardBadge, {
      props: {
        asset,
      },
    })

    expect(component.html()).toMatchSnapshot()
  })
})
