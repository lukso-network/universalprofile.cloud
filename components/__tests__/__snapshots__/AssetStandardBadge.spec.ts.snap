// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AssetStandardBadge > should render LSP7 badge 1`] = `
"<div id="test-wrapper">
  <div class="flex whitespace-nowrap">
    <lukso-tag size="x-small" background-color="lukso-90">LSP7</lukso-tag>
  </div>
</div>"
`;

exports[`AssetStandardBadge > should render LSP8 badge 1`] = `
"<div id="test-wrapper">
  <div class="flex whitespace-nowrap">
    <lukso-tag size="x-small" background-color="lukso-90">LSP8</lukso-tag>
  </div>
</div>"
`;

exports[`AssetStandardBadge > should render collection badge 1`] = `
"<div id="test-wrapper">
  <div class="flex whitespace-nowrap">
    <lukso-tag size="x-small" background-color="lukso-90">LSP8 COLLECTION</lukso-tag>
  </div>
</div>"
`;

exports[`AssetStandardBadge > should render loading state 1`] = `
"<div id="test-wrapper">
  <div class="animate-pulse rounded-4 bg-neutral-90 h-[20px] w-10"></div>
</div>"
`;

exports[`AssetStandardBadge > should render slot content 1`] = `
"<div id="test-wrapper">
  <lukso-tag size="x-small" background-color="lukso-90">Slot content</lukso-tag>
</div>"
`;
